apps/meteor/server/services/team/service.ts

```diff
@@ -137,7 +138,14 @@ export class TeamService extends ServiceClassInternal implements ITeamService {
 				_id: teamId,
 				...teamData,
 			};
-		} catch (e) {
+		} catch (error) {
+			// Manually deleting team if an error occurs, this is a temporary fix due to urgency
+			// to release version 6.12 and will be refactored on next release.
+			if (teamId) {
+				await Team.deleteOneById(teamId);
+				await TeamMember.deleteByTeamId(teamId);
+			}
+
```

Comment:
What if we create the team only after the rooms has been succesfully created? This way we wouldn't need any DB calls in the catch statement

Reply:
I think the room is created with teamId iirc

Reply:
> I think the room is created with teamId iirc

We can update the room's `teamId` afterwards.. I've added this change as well.